libraries/lib-files/PlatformCompatibilityMacOS.cpp

```diff
@@ -0,0 +1,107 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  PlatformCompatibility.cpp
+
+  Markus Meyer
+
+*******************************************************************//*!
+
+\class PlatformCompatibility
+\brief Filename Compatibility utilities.
+
+\see FileNames
+
+*//*******************************************************************/
+
+#include "PlatformCompatibility.h"
+#include <CoreServices/CoreServices.h>
+#include <limits.h>
+#include <mach-o/dyld.h>
+#include <pwd.h>
+#include <string>
+#include <sys/types.h>
+#include <unistd.h>
+
+std::string PlatformCompatibility::GetUserDataDir()
```

Comment:
For us, these classes that return paths can become a problem.
The application as a whole should have one source of truth for returning standart paths.
In AU4, this is IGlobalConfiguration (https://github.com/musescore/framework_tmp/blob/main/framework/global/iglobalconfiguration.h) (internally, this is QStandardPaths)
If we use these classes, there will be two sources of paths and the paths can be different for the same entity, this will become a problem.
It would be good to do it like this:
* In AU3, add an interface that returns paths.
* Add the ability to register an implementation of this interface.
* In AU4, we register our implementation, which receives paths from IGlobalConfiguration.
* In AU3, our own implementation should be substituted, if needed.

Reply:
I agree, there should be one source of truth. If you already have this functionality implemented in AU4, that's the one that should be reused, of course.
This code is a bit outdated. @vsverchinsky suggested a better approach which I applied to my PR: see [this commit](https://github.com/audacity/audacity/pull/7017/commits/17d74067d6794f81ee12cd2ab4483e4bde1704e1). In the AU3 source tree, we add `PlatformCompatibilityWX.cpp`, which provides definitions for `PlatformCompatibility::GetUserDataDir()` and the rest. I had in mind that the au4 sources could have `PlatformCompatibilityQt.cpp`, which would invoke `GlobalConfiguration::userDataPath()` and other appropriate methods?

Reply:
> I had in mind that the au4 sources could have PlatformCompatibilityQt.cpp, which would invoke GlobalConfiguration::userDataPath() and other appropriate methods?

Just please let's make sure that there is no Qt in the name :) 
The following options are possible:
* PlatformCompatibilityAU4
* PlatformCompatibilityMuse